google/cloud/spanner/mutations.h

```diff
@@ -193,12 +191,50 @@ Mutation MakeInsertMutation(Ts&&... values) {
 using UpdateMutationBuilder =
     internal::WriteMutationBuilder<internal::UpdateOp>;
 
-/// Creates a simple insert mutation for the values in @p row.
+/// Creates a simple update mutation for the values in @p row.
 template <typename... Ts>
 Mutation MakeUpdateMutation(Ts&&... values) {
   return UpdateMutationBuilder().AddRow(std::forward<Ts>(values)...).Build();
 }
 
+/**
+ * A helper class to construct "insert_or_update" mutations.
+ *
+ * @see The Mutation class documentation for an overview of the Cloud Spanner
+ *   mutation API
+ *
+ * @see https://cloud.google.com/spanner/docs/modify-mutation-api
+ *   for more information about the Cloud Spanner mutation API.
+ */
+using InsertOrUpdateMutationBuilder =
+    internal::WriteMutationBuilder<internal::InsertOrUpdateOp>;
+
+/// Creates a simple "insert or update" mutation for the values in @p row.
+template <typename... Ts>
+Mutation MakeInsertOrUpdateMutation(Ts&&... values) {
+  return InsertOrUpdateMutationBuilder()
+      .AddRow(std::forward<Ts>(values)...)
+      .Build();
+}
+
+/**
+ * A helper class to construct "insert_or_update" mutations.
+ *
+ * @see The Mutation class documentation for an overview of the Cloud Spanner
+ *   mutation API
+ *
+ * @see https://cloud.google.com/spanner/docs/modify-mutation-api
+ *   for more information about the Cloud Spanner mutation API.
+ */
+using ReplaceMutationBuilder =
+    internal::WriteMutationBuilder<internal::ReplaceOp>;
+
+/// Creates a simple "replace" mutation for the values in @p row.
```

Comment:
The argument name is `values`. But maybe I'm confused about doxygen's `@p` directive.

Reply:
Fixed.